.circleci/config.yml

```diff
@@ -592,7 +617,17 @@ gke_test: &gke_test
     - run:
         name: Export escaped GKE cluster name
         command: |
-          echo "export ESCAPED_GKE_CLUSTER=$(echo ${GKE_CLUSTER}-${CIRCLE_BRANCH:-$CIRCLE_TAG}-${TEST_LATEST_RELEASE}-${GKE_BRANCH}-ci | sed 's/[^a-z0-9-]//g')" >> $BASH_ENV
+          ESCAPED_GKE_CLUSTER=$(echo ${GKE_CLUSTER}-${CIRCLE_BRANCH:-$CIRCLE_TAG}-${TEST_LATEST_RELEASE}-${GKE_BRANCH}-ci | sed 's/[^a-z0-9-]//g')
+          
+          # Trim the cluster name in case it exceeds the max length allowed by GKE (currently 40 chars)
+          MAX_LENGTH=40 
+          len=${#ESCAPED_GKE_CLUSTER}
+          if ((len > MAX_LENGTH)); then
+            ESCAPED_GKE_CLUSTER="${ESCAPED_GKE_CLUSTER:0:MAX_LENGTH}"
```

Comment:
Given that we might be trimming the name, perhaps we should move the `GKE_BRANCH` close to the beginning of the name?  I mean, to avoid name collisions if the name is trimmed out, we may want to have the disambiguating part first.

Reply:
yeah, you're right, although for the CircleCI flow I haven't added the `gke_release_channel` param, so in case `stable` and `regular` versions are the same, as they are right now, they will collide anyway...